volatility3/framework/plugins/linux/proc.py

```diff
@@ -35,16 +40,138 @@ def get_requirements(cls):
+                default=False,
+                optional=True,
+            ),
+            requirements.ListRequirement(
+                name="address",
+                description="Process virtual memory address to include "
+                "(all other address ranges are excluded). This must be "
+                "a base address, not an address within the desired range.",
+                element_type=int,
+                optional=True,
+            ),
+            requirements.IntRequirement(
+                name="maxsize",
+                description="Maximum size for dumped VMA sections "
+                "(all the bigger sections will be ignored)",
+                default=cls.MAXSIZE_DEFAULT,
+                optional=True,
+            ),
         ]
 
+    @classmethod
+    def list_vmas(
+        cls,
+        task: interfaces.objects.ObjectInterface,
+        filter_func: Callable[
+            [interfaces.objects.ObjectInterface], bool
+        ] = lambda _: False,
+    ) -> Generator[interfaces.objects.ObjectInterface, None, None]:
+        """Lists the Virtual Memory Areas of a specific process.
+
+        Args:
+            task: task object from which to list the vma
+            filter_func: Function to take a vma and return True if it should be filtered out
+
+        Returns:
+            A list of vmas based on the task and filtered based on the filter function
+        """
+        if task.mm:
+            for vma in task.mm.get_mmap_iter():
+                if not filter_func(vma):
+                    yield vma
+
+    @classmethod
+    def vma_dump(
+        cls,
+        context: interfaces.context.ContextInterface,
+        task: interfaces.objects.ObjectInterface,
+        vma: interfaces.objects.ObjectInterface,
+        open_method: Type[interfaces.plugins.FileHandlerInterface],
+        maxsize: int = MAXSIZE_DEFAULT,
+    ) -> Optional[interfaces.plugins.FileHandlerInterface]:
+        """Extracts the complete data for VMA as a FileInterface.
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            task: an task_struct instance
+            vma: The suspected VMA to extract (ObjectInterface)
+            open_method: class to provide context manager for opening the file
+            maxsize: Max size of VMA section (default MAXSIZE_DEFAULT)
+
+        Returns:
+            An open FileInterface object containing the complete data for the task or None in the case of failure
+        """
+        try:
+            vm_start = vma.vm_start
+            vm_end = vma.vm_end
+        except AttributeError:
+            vollog.debug("Unable to find the vm_start and vm_end")
+            return None
+        
+        vm_size = vm_end - vm_start
+        if 0 < maxsize < vm_size:
+            vollog.debug(
+                f"Skip virtual memory dump {vm_start:#x}-{vm_end:#x} due to maxsize limit"
```

Comment:
Should this skip, or return up to maxsize?  I don't mind either, I'd just like it clear so as not to confuse people why things didn't return (particularly since the warning requires debug verbosity to have been turned on to see it).